pkg/dpll/dpll.go

```diff
@@ -649,20 +650,24 @@ func (d *DpllConfig) stateDecision() {
 		} else { // what happens if source is lost and DPLL is locked? goto holdover?
 			glog.Infof("dpll is locked, offset is out of range, state is FREERUN(%s)", d.iface)
 			d.state = event.PTP_FREERUN
+			d.phaseOffset = FaultyPhaseOffset
 		}
 		d.inSpec = true
 		d.sendDpllEvent()
 	case DPLL_LOCKED_HO_ACQ, DPLL_HOLDOVER:
 		if d.hasPPSAsSource() {
 			if dpllStatus == DPLL_HOLDOVER {
 				d.state = event.PTP_FREERUN // pps when moved to HOLDOVER we declare freerun
+				d.phaseOffset = FaultyPhaseOffset
 				d.sourceLost = true
 			} else if dpllStatus == DPLL_LOCKED_HO_ACQ && d.isOffsetInRange() {
 				d.state = event.PTP_LOCKED
 				d.sourceLost = false
+				d.inSpec = true
 			} else {
```

Comment:
dpll Status ins in LOCKED_HO_ACQ here you do get offsets ( set offset to faulty in case where you don't get offsets )